internal/credentials/spiffe/spiffe_test.go

```diff
@@ -0,0 +1,129 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package spiffe
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"io"
+	"os"
+	"testing"
+
+	"google.golang.org/grpc/testdata"
+)
+
+func TestKnownSPIFFEBundle(t *testing.T) {
+	spiffeBundleFile := testdata.Path("spiffe/spiffebundle.json")
+	bundles, err := LoadSPIFFEBundleMap(spiffeBundleFile)
+	if err != nil {
+		t.Fatalf("LoadSPIFFEBundleMap(%v) Error during parsing: %v", spiffeBundleFile, err)
+	}
+	wantBundleSize := 2
+	if len(bundles) != wantBundleSize {
+		t.Fatalf("LoadSPIFFEBundleMap(%v) did not parse correct bundle length. got %v want %v", spiffeBundleFile, len(bundles), wantBundleSize)
+	}
+	if bundles["example.com"] == nil {
+		t.Fatalf("LoadSPIFFEBundleMap(%v) got no bundle for example.com", spiffeBundleFile)
+	}
+	if bundles["test.example.com"] == nil {
+		t.Fatalf("LoadSPIFFEBundleMap(%v) got no bundle for test.example.com", spiffeBundleFile)
+	}
+
+	expectedExampleComCert := loadX509Cert(t, testdata.Path("spiffe/spiffe_cert.pem"))
+	expectedTestExampleComCert := loadX509Cert(t, testdata.Path("spiffe/server1_spiffe.pem"))
+	if !bundles["example.com"].X509Authorities()[0].Equal(expectedExampleComCert) {
+		t.Fatalf("LoadSPIFFEBundleMap(%v) parsed wrong cert for example.com.", spiffeBundleFile)
+	}
+	if !bundles["test.example.com"].X509Authorities()[0].Equal(expectedTestExampleComCert) {
+		t.Fatalf("LoadSPIFFEBundleMap(%v) parsed wrong cert for test.example.com", spiffeBundleFile)
+	}
+
+}
+
+func loadX509Cert(t *testing.T, filePath string) *x509.Certificate {
+	t.Helper()
+	certFile, _ := os.Open(filePath)
+	certRaw, _ := io.ReadAll(certFile)
+	block, _ := pem.Decode([]byte(certRaw))
+	if block == nil {
+		t.Fatalf("pem.Decode(%v) = nil. Want a value.", certRaw)
+	}
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		t.Fatalf("x509.ParseCertificate(%v) failed %v", block.Bytes, err.Error())
+	}
+	return cert
+}
+
+func TestLoadSPIFFEBundleMapFailures(t *testing.T) {
+	testCases := []struct {
+		filePath string
+	}{
+		{
+			filePath: testdata.Path("spiffe/spiffebundle_corrupted_cert.json"),
+		},
+		{
+			filePath: testdata.Path("spiffe/spiffebundle_malformed.json"),
+		},
+		{
+			filePath: testdata.Path("spiffe/spiffebundle_wrong_kid.json"),
+		},
+		{
+			filePath: testdata.Path("spiffe/spiffebundle_wrong_kty.json"),
+		},
+		{
+			filePath: testdata.Path("spiffe/spiffebundle_wrong_multi_certs.json"),
+		},
+		{
+			filePath: testdata.Path("spiffe/spiffebundle_wrong_root.json"),
+		},
+		{
+			filePath: testdata.Path("spiffe/spiffebundle_wrong_seq_type.json"),
+		},
+		{
+			filePath: testdata.Path("NOT_A_REAL_FILE"),
+		},
+		{
+			filePath: testdata.Path("spiffe/spiffebundle_invalid_trustdomain.json"),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.filePath, func(t *testing.T) {
+			_, err := LoadSPIFFEBundleMap(tc.filePath)
+			if err == nil {
```

Comment:
Nit: The assignment and the conditional can be on a single line.

Reply:
Done